include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -156,10 +156,14 @@ compared to 1*64 when the hasLeadingOffset is false.
             //
             // maxPhase is set to SIMDWidth / perPhase
             int vecSize = ((typeWidthInBit == 16) ? 64 : 32 ) / typeWidthInBit;
-            int maxPhase = SIMDWidth / perPhase;
+            // On MI300, fp8 and int8 mfma is assigned to VGPRs for the operands
+            if (3 == mfmaEnc.getVersionMajor() && 8 == typeWidthInBit)
```

Comment:
I don't quite know what `mfmaEnc.getVersionMajor()` is for but sounds like for 8bit and 16bit it uses 2 VGPR and for 32bit (and maybe 64?) it uses 1 VGPR. I'll trust you did the correct thing as I don't have enough experience on this but it can have nicer structure as more rules come.

Also, not important but, why it is written as `3 == mfmaEnc.getVersionMajor` rather than the other way around?

Reply:
Your understanding is correct. 
The tricky thing is, when versionMajor == 2 or 1, 8bit uses 1 VGPR.

You are also right this is a little hard to understand. I promise I'll clean up the logic later.

3 == x is just a trick to avoid the possibility of writing it as 3 = x, in which case the code won't compile.